itest.py

```diff
@@ -0,0 +1,832 @@
+
+        if failed_tests_list:
+            running_arg_list = arg_list + failed_tests_list
+        else:
+            running_arg_list = arg_list
+
+        logger.debug(f"Executing running_arg_list=[{running_arg_list!r}]")
+
+        try:
+            #with subprocess.Popen(running_arg_list, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True) as res:
+            #    for line in res.stdout:
+            #        print(line, end='')
+
+            #res = stream_subprocess_output(running_arg_list, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True)
+            res = subprocess.run(running_arg_list, capture_output=True, text=True)
+        except KeyboardInterrupt:
+            logger.warning('Interupt recieved. Kill test...')
+            arg_list = []
+            d_c = shutil.which('docker-compose')
+            arg_list.append(d_c)
+            arg_list.append('down')
+            subprocess.run(arg_list, cwd=str(project_directory))
+            c = input('Continue? (y/N)')
+            if c.lower() == 'n' or len(c) == 0:
+                break
+            else:
+                continue
+
+        logger.debug(f'res=[{res!r}] of attempt=[{attempt!r}]')
+
+        if res.returncode == 0:
+            sha = SHA_RE.search(res.stdout).group('sha')
+            logger.info(f'Test passed on attempt=[{attempt!r}], sha=[{sha!r}]')
+            break
+
+        failed_tests = failed_tests_pattern.search(res.stdout)
+
+        if failed_tests:
+            new_failed_tests_list = failed_tests.group().strip().split()
+            new_failed_tests_list.insert(0, '--tests')
+            if new_failed_tests_list == failed_tests_list:
+                current_duplicate_attempt += 1
+            else:
+                current_duplicate_attempt = 1
+            failed_tests_list = new_failed_tests_list
+        else:
+            logger.error(f'Failed to parse failed_tests with failed_tests_pattern=[{failed_tests_pattern!r}]')
+            current_duplicate_attempt += 1
+
+        logger.info(f'attempt=[{attempt!r}] failed')
+
+    return res
+
+def ask_run_compile(os=None, build_args=None):
+    logger.debug(f'ask_run_compile(os=[{os!r}], build_args=[{build_args!r}])')
+    client = docker.from_env()
+
+    yes_images = tuple(get_valid_tags(client, IRODS_BUILD_IMAGE_NAME))
+    assert(len(yes_images) > 0)
+
+    if os is None:
+        for image in yes_images:
+            print(image)
+
+        os_choice = input("What image ya wanna run? ")
+    else:
+        os_choice = os
+
+    validate_choice(os_choice, yes_images)
+
+    build_image = client.images.get(f'{IRODS_BUILD_IMAGE_NAME}:{os_choice}')
+    
+    dirs_to_format = [IRODS_SERV_BUILD_DIR, IRODS_ICOMMANDS_BUILD_DIR, IRODS_PACKAGES_DIR, IRODS_CCACHE_DIR]
+    dirs_to_mount = [d for d in format_pure_path(dirs_to_format, os_name=os_choice)]
+
+    # Cleanup build dirs
+    # shutil.rmtree(dirs_to_mount[0])
+    # shutil.rmtree(dirs_to_moint[1])
+
+    # make_dirs_if_needed()
```

Comment:
Remove dead code, add issues for:
- Cleanup of build dirs
- Funciton for creation of needed dirs

Create issues only if desired.

Reply:
Dead code removed. Resolving, we can add these issues later if really needed.